SpringProject/src/it/java/com/gradproject2019/auth/AuthControllerIT.java

```diff
@@ -94,21 +105,55 @@ public void shouldReturn401WhenUserExistsAndPasswordIncorrect() {
 
         //when
         userRepository.saveAndFlush(user);
-        ResponseEntity<ErrorEntity> response = loginExpectingError(loginUri, request);
+        ResponseEntity<ErrorEntity> response = loginExpectingError(request);
 
         //then
         Assert.assertEquals(401, response.getStatusCodeValue());
         Assert.assertEquals("Invalid user credentials.", response.getBody().getMessage());
     }
 
-    private ResponseEntity<Token> login(URI loginUri, HttpEntity<LoginDto> request) {
-        return restTemplate.exchange(loginUri, POST, request, Token.class);
+    //test to check the delete token endpoint
+    @Test
+    public void shouldReturn204andDeleteTokenWhenTokenExists() {
+        //given token exists
+        userRepository.saveAndFlush(user);
```

Comment:
If you've already saved a user in your before method, you shouldn't need to save one here